core/src/test/java/dk/alexandra/fresco/lib/lp/LpBuildingBlockTests.java

```diff
@@ -0,0 +1,602 @@
+          System.out.println(output);
+          assertTrue(output
+              .contains("C: \n" + "0, 0, 0, \n" + "1, 1, 1, \n" + "2, 2, 2, \n" + "3, 3, 3,"));
+          assertTrue(output.contains("B: \n" + "10, 10, 10, 10,"));
+          assertTrue(output.contains("F: \n" + "0, 2, 3,"));
+          assertTrue(output.contains("z: \n" + "0"));
+        }
+      };
+    }
+  }
+
+  public static class TestLpSolver<ResourcePoolT extends ResourcePool> 
+  extends TestThreadFactory <ResourcePoolT, ProtocolBuilderNumeric> {
+
+    private LPSolver.PivotRule pivotRule; 
+    
+    public TestLpSolver(LPSolver.PivotRule pivotRule) {
+      this.pivotRule = pivotRule;
+    }
+
+    @Override
+    public TestThread<ResourcePoolT, ProtocolBuilderNumeric> next() {
+      return new TestThread<ResourcePoolT, ProtocolBuilderNumeric>() {
+
+        @Override
+        public void test() throws Exception {
+          LpSolverTester app = new LpSolverTester() {
+
+            @Override
+            public DRes<BigInteger> buildComputation(ProtocolBuilderNumeric builder) {
+              return setup(builder, pivotRule);
+            }
+          };
+          BigInteger out = runApplication(app);
+          assertEquals(app.getExpectedOptimal(), out);
+        }
+      };
+    }
+  }
+  
+  public static class TestLpSolverDebug<ResourcePoolT extends ResourcePool>
+      extends TestThreadFactory<ResourcePoolT, ProtocolBuilderNumeric> {
+
+    public TestLpSolverDebug() {}
+
+    @Override
+    public TestThread<ResourcePoolT, ProtocolBuilderNumeric> next() {
+      return new TestThread<ResourcePoolT, ProtocolBuilderNumeric>() {
+
+        @Override
+        public void test() throws Exception {
+          LpSolverDebugTester app = new LpSolverDebugTester() {
+
+            @Override
+            public DRes<BigInteger> buildComputation(ProtocolBuilderNumeric builder) {
+              return setup(builder, LPSolver.PivotRule.DANZIG);
+            }
+          };
+          if (this.conf.getMyId() == 1) {
+            ByteArrayOutputStream out = new ByteArrayOutputStream();
+            PrintStream stdout = System.out;
+            System.setOut(new PrintStream(out));
+            runApplication(app);
+            String s = out.toString();
+            System.setOut(stdout);
+            System.out.println(s);
+            s.replaceAll("\r", "");
+            assertTrue(s.contains("C:"));
+            assertTrue(s.contains(
+                "C: \n" + "1, 0, 0, 1, 0, 0, \n" + "0, 1, 0, 0, 1, 0, \n" + "0, 0, 1, 0, 0, 1,"));
+            assertTrue(s.contains("B: \n" + "1, 2, 3, "));
+            assertTrue(s.contains("F: \n" + "-1, -1, -1, 0, 0, 0, "));
+            assertTrue(s.contains("z: \n" + "0"));
+            assertTrue(s.contains("Basis [1]: \n" + "4, 5, 6,"));
+            assertTrue(s.contains("Basis [4]: \n" + "1, 2, 3,"));
+            assertTrue(s.contains("Update Matrix [1]: \n" + "1, 0, 0, 0, \n" + "0, 1, 0, 0, \n"
```

Comment:
Ditto - consider recoding tests on output stream so they have a input, key and an expected value

Reply:
As above.